### PR TITLE
Change the test code to only check for 4 significant figures.

### PR DESCRIPTION
**Pull request by mithro**
This fixes a large number of the tests which were disabled for "floating point" errors in firefox.

### CODE REVIEW

File: test/bootstrap.js [hunk 1]
```diff
@@ -221,6 +220,20 @@ function _assert_important_in_array(actual, expected, message) {
 
       try {
         assert_equals(typeof actual, typeof expected);
+
+        if (typeof actual === "number") {
```
Comment (dstoc) [inline test/bootstrap.js]: single quotes

File: test/bootstrap.js [hunk 2]
```diff
@@ -221,6 +220,20 @@ function _assert_important_in_array(actual, expected, message) {
...
+          if (Math.abs(actual) < 1e-10) {
+            actual = 0;
+          }
+          actual = ""+actual.toPrecision(4);
```
Comment (dstoc) [inline test/bootstrap.js]: spaces around binary operators

File: test/test-generator.html [hunk 3]
```diff
@@ -206,6 +206,14 @@
         }
         console.log("Node:", checks[x].object[i].cloneNode(), "Property:", checks[x].properties[j], "Value:", value);
 
+        value = value.replace(/[0-9.\-eE]+/g, function(v) {
```
Comment (dstoc) [inline test/test-generator.html]: Do we really need to change the generator?

File: test/bootstrap.js [hunk 4]
```diff
@@ -221,6 +220,20 @@ function _assert_important_in_array(actual, expected, message) {
...
+        }
+        if (typeof expected === "number") {
+          if (Math.abs(expected) < 1e-10) {
+            expected = 0;
+          }
+          expected = ""+expected.toPrecision(4);
```
Comment (dstoc) [inline test/bootstrap.js]: why are we converting to strings?

File: test/bootstrap.js (hunk 1 above)
Comment (mithro) [inline test/bootstrap.js]: Fixed. (One of these days I'll get jslint working and these will all go away...)

File: test/bootstrap.js (hunk 2 above)
Comment (mithro) [inline test/bootstrap.js]: Fixed. (One of these days I'll get jslint working and these will all go away...)

File: test/test-generator.html (hunk 3 above)
Comment (mithro) [inline test/test-generator.html]: It makes the values much nicer to read and smaller too.

File: test/testcases/auto-test-composite-transforms-checks.js [hunk 5]
```diff
@@ -1,50 +1,45 @@
 timing_test(function() {
   at(0, function() {
-    assert_styles(
-      '.anim',
-      [{'transform':'matrix(1, 0, 0, 1, 0, 0)'},
-       {'transform':'matrix(1, 0, 0, 1, 0, 0)'},
-       {'transform':'matrix(1, 0, 0, 1, 0, 0)'},
-       {'transform':'matrix(1, 0, 0, 1, 0, 0)'}]);
-  }, "Autogenerated");
+    assert_styles(".anim",{'transform':'matrix(1.000, 0.000, 0.000, 1.000, 0.000, 0.000)'});
```
Comment (dstoc) [inline test/testcases/auto-test-composite-transforms-checks.js]: be nice if these were normalized back to 0 and 1

File: test/test-generator.html (hunk 3 above)
Comment (dstoc) [inline test/test-generator.html]: Right, after I got down to the tests, I agree.

**Comment by dstoc** [comment]
lgtm

File: test/bootstrap.js (hunk 4 above)
Comment (mithro) [inline test/bootstrap.js]: Because strings compare reliably.

File: test/testcases/auto-test-composite-transforms-checks.js (hunk 5 above)
Comment (mithro) [inline test/testcases/auto-test-composite-transforms-checks.js]: We should just strip off 0's after the decimal point? Something like;

s/.?0+$//

Comment (mithro) [inline test/testcases/auto-test-composite-transforms-checks.js]: Done.

**Comment by shans** [comment]
This should have gotten an additional review after commits df2ab and 881e3.

881e3 removed a number of useful tests (auto-test-compositor .anim1 x tests were converted to .anim1 ctm tests). This would most likely have been picked up in review.

Fixed by #486

**Pull request merged**